ios/CoinbaseWalletSDK/CoinbaseWalletSDK.swift

```diff
@@ -16,63 +16,56 @@ public final class CoinbaseWalletSDK {
         return UIApplication.shared.canOpenURL(URL(string: "cbwallet://")!)
     }
     
-    // MARK: - Constructor
-    
-    static private var host: URL?
     static private var callback: URL?
     static public var isConfigured: Bool {
-        return host != nil && callback != nil
+        return callback != nil
     }
     
-    static public func configure(
-        host: URL = URL(string: "https://wallet.coinbase.com/wsegue")!,
-        callback: URL
-    ) {
-        guard isConfigured == false else {
-            assertionFailure("`CoinbaseWalletSDK.configure` should be called only once.")
-            return
-        }
-        
-        self.host = host
+    static public func configure(callback: URL) {
         if callback.pathComponents.count < 2 { // [] or ["/"]
             self.callback = callback.appendingPathComponent("wsegue")
         } else {
             self.callback = callback
         }
     }
     
-    static public var shared: CoinbaseWalletSDK = {
-        guard let host = CoinbaseWalletSDK.host,
-              let callback = CoinbaseWalletSDK.callback else {
-            preconditionFailure("Missing configuration: call `CoinbaseWalletSDK.configure` before accessing the `shared` instance.")
+    static public func getInstance(hostWallet: Wallet) -> CoinbaseWalletSDK? {
+        guard let callback = callback else {
+            assertionFailure("`CoinbaseWalletSDK.configure` should be called prior to retrieving an instance.")
+            return nil
         }
         
-        return CoinbaseWalletSDK(host: host, callback: callback)
-    }()
+        let host = URL(string: hostWallet.url)!
+        if (instances[host] == nil) {
+            let newInstance = CoinbaseWalletSDK(host, callback)
+            instances[host] = newInstance
+        }
+        
+        return instances[host]!
+    }
     
+    private static var instances: [URL: CoinbaseWalletSDK] = [:]
     // MARK: - Properties
     
     private let appId: String
-    private let host: URL
-    private let callback: URL
+    private let hostURL: URL
+    private let callbackURL: URL
     
     private lazy var keyManager: KeyManager = {
-        KeyManager(host: self.host)
-    }()
-    private lazy var taskManager: TaskManager = {
-        TaskManager()
+        KeyManager(host: self.hostURL)
     }()
     
-    private init(
-        host: URL,
-        callback: URL
-    ) {
-        self.host = host
-        self.callback = callback
+    private init(_ host: URL,_ callback: URL) {
```

Comment:
these params both have URL types. let's keep tags
```suggestion
    private init(host: URL, callback: URL) {
```